controllers/tmf-apis/catalog.js

```diff
@@ -742,6 +742,26 @@ var catalog = (function() {
         }
     };
 
+    var isEqualRelatedParty = function (relatedParty1, relatedParty2) {
+
+        var equal = true;
+
+        if (relatedParty1.length != relatedParty2.length) {
+            equal = false;
+        } else {
+
+            for (var i = 0; i < relatedParty1.length && equal; i++) {
+                if (relatedParty1[i].id !== relatedParty2[i].id ||
+                    relatedParty1[i].href !== relatedParty2[i].href ||
+                    relatedParty1[i].role !== relatedParty2[i].role) {
+                    equal = false;
+                }
+            }
```

Comment:
This way of checking is supposing that the server is always giving the relatedParties in the same order. If the server changes the order in the list when retrieving the previous body, the request will fail event if the user has not changed the related party field